cloudpickle/cloudpickle.py

```diff
@@ -707,7 +707,7 @@ def _function_getstate(func):
         # Hack to circumvent non-predictable memoization caused by string interning.
         # See the inline comment in _class_setstate for details.
         "__name__": "".join(func.__name__),
-        "__qualname__": func.__qualname__,
+        "__qualname__": "".join(func.__qualname__),
```

Comment:
It was previously not interned but now it is without rebuilding the string?
Not super clear why this behavior changed, I could not find any lead in the python3.13 release note.

But if it does the trick, this seems fine like this, this should not be too costly to do.